kqueen/blueprints/api/views.py

```diff
@@ -220,6 +220,27 @@ def cluster_resize(pk):
     return jsonify(output)
 
 
+@api.route('/clusters/<uuid:pk>/set_network_policy', methods=['PATCH'])
+@jwt_required()
+def cluster_set_network_policy(pk):
+    obj = get_object(Cluster, pk, current_identity)
+
+    data = request.json
+    if not isinstance(data, dict) or (isinstance(data, dict) and 'node_count' not in data):
+        abort(400)
+
+    res_status, res_msg = obj.engine.set_network_policy(data['network_policy']['provider'],
+                                                        data['network_policy']['enabled'])
+
+    if not res_status:
+        logger.error('Setting network policy failed: {}'.format(res_msg))
+        abort(500, description=res_msg)
```

Comment:
Actually, I don't think that 500 error is a good choice here. It indicates with problems on server side, but if this action is unappropriate due to he recommended minimum size \
+                   cluster to run network policy enforcement is 3 \
+                   n1-standard-1 instances or small node count, the respond code should be forbidden.

Reply:
and what do u prefer?

Reply:
when user doesn't satisfy cluster to run network policy enforcement is 3  n1-standard-1 instances or 2 node count - 403 -forbidden, should be return, not 500. you can return it in set_network_policy as part of tuple that you are already returning